content/blog/2021/07/2021-07-29-introducing-conventional-commits-plugin-for-jenkins.adoc

```diff
@@ -0,0 +1,121 @@
+:sig: gsoc
+opengraph:
+  image: /images/gsoc/2021/conventional-commits/conventionalCommitsPluginForJenkins.png
+links:
+  discourse: https://community.jenkins.io/t/introducing-the-conventional-commits-plugin/272
+---
+
+image:/images/gsoc/2021/conventional-commits/conventionalCommitsPluginForJenkins.png[GSoC, height=420, role=center, float=center]
+
+The conventional commits plugin is Google Summer of Code project. Special thanks to the mentors Gareth Evans, Kristin Whetstone, Olivier Vernin and Alan Buradezwic 
+
+=== What are Conventional Commits
+
+According to the link:https://www.conventionalcommits.org/[official website], conventonal commits are, "A specification for adding human and machine readable meaning to commit messages."
+
+Conventional commits are a lightweight convention on top of commit messages.
+
+The following table shows major structural elements offered by the conventional commits convention.
+
+|===
+|Structural Element|Example
+
+|Chore
+|`chore: improve logging`
+
+|Fix
+|`fix: minor bug fix`
+
+|Feat
+|`feat: add a new feature`
+
+|Breaking Change
+|`BREAKING CHANGE: reimplement`
+|===
+
+=== Why Conventional Commits
+
+As the CI/CD world is moving more towards complete automation and minimal human interaction, the ability to fully automate a release is desired.
+Conventional Commits enable the use of automated systems on top of commit messages.
+These systems can "truly" automate a release with almost no human interaction. 
+
+The convention dovetails with link:https://semver.org/[SemVer]. Let's take an example, a maven project is currently versioned at `1.2.0`. The following table shows how conventional commits would bump the verison depending on the type of the commit.
+
+|===
+|Commit Message|Version Bump|SemVer Equivalent
+
+|`chore: improve logging`
+|`1.2.0` -> `1.2.0`
+|No version bump
+
+|`fix: minor bug fix`
+|`1.2.0` -> `1.2.1`
+|Increament in the patch version
+
+|`feat: add a new feature`
+|`1.2.0` -> `1.3.0`
+|Increment in the minor version
+
+|`BREAKING CHANGE: reimplement`
+|`1.2.0` -> `2.0.0`
+|Increment in the major version
+|===
+
+
+=== The Conventional Commits Plugin
+
+The conventional commits plugin is a Jenkins plugin to programatically determine the next semantic version of a git repository using:
+
+- Last tagged version 
+- Commit message log
+
+==== How it works?
+The plugin will read the commit messages from the lastest tag till the lastest commit and determine what would be the next (Semantic) Version for that particular project.
+
+==== Supported Project Types?
+Currently the plugin supports the following project types:
+
+- Maven
+- Gradle
+- Make
+- Python
+- Helm
+- Node (NPM)
+
+==== How to request a project type support? 
+Please feel free to open an issue on the link:https://github.com/jenkinsci/conventional-commits-plugin/[GitHub repository] of the plugin.
+
+=== How to use the plugin
+
+Recommended way of using the plugin is to add a step in a Jenkins Pipeline Project.
```

Comment:
Could you insert the example step here so that the reader can see it?
```suggestion
Recommended way of using the plugin is to add a step in a Jenkins Pipeline Project.
```

Reply:
Oh yes sure. Thank you for the suggestion 😄